sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -220,10 +235,24 @@ object FileFormat {
           // the modificationTime from the file is in millisecond,
           // while internally, the TimestampType `file_modification_time` is stored in microsecond
           row.update(i, fileModificationTime * 1000L)
+        case ROW_INDEX =>
+          // Reserve the spot in the row for a LongType value. The metadata fields that have
+          // identical values for each row of the file are set by this function, while fields that
+          // have different values (such as row index) are set separately.
+          row.update(i, -1L)
       }
     }
     row
   }
+
+  /**
+   * Does the given metadata column always contain identical values for all rows originating from
```

Comment:
nit: could this be rephrased as a statement instead? Thanks. Otherwise, people might confuse it with a TODO item.

Reply:
Done.